diff --git a/lib/src/facts/collection.cc b/lib/src/facts/collection.cc
index 9f2859f..bc073f4 100644
--- a/lib/src/facts/collection.cc
+++ b/lib/src/facts/collection.cc
@@ -53,7 +53,7 @@ namespace facter { namespace facts {
     void collection::add_default_facts()
     {
         add("cfacterversion", make_value<string_value>(LIBFACTER_VERSION));
-        add_platform_facts();
+        //add_platform_facts();
     }
 
     void collection::add(shared_ptr<resolver> const& res)
