diff --git a/CMakeLists.txt b/CMakeLists.txt
index e54e9b5..16dafae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,9 +41,9 @@ endif()
# not set CMAKE_CXX_FLAGS globally because gtest is not warning-clean.
if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
    # Clang warns that 'register' is deprecated; 'register' is used throughout boost, so it can't be an error yet.
    set(FACTER_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall [--Wextra-]{+-Wno-missing-field-initializers -Wno-unused-variable+} -Werror -Wno-unused-parameter -Wno-tautological-constant-out-of-range-compare -Wno-deprecated-register")
elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
    set(FACTER_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall [--Wextra-]{+-Wno-missing-field-initializers -Wno-unused-variable+} -Werror -Wno-unused-parameter -Wno-unused-local-typedefs -Wno-unknown-pragmas")
elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
endif()
diff --git a/lib/tests/CMakeLists.txt b/lib/tests/CMakeLists.txt
index 7138cd6..0c1f105 100644
--- a/lib/tests/CMakeLists.txt
+++ b/lib/tests/CMakeLists.txt
@@ -4,9 +4,9 @@ find_package(Ruby)

# Set compiler-specific flags
if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall [--Wextra-]{+-Wno-missing-field-initializers -Wno-unused-variable+} -Werror -Wno-unused-parameter -Wno-tautological-constant-out-of-range-compare")
elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall [--Wextra-]{+-Wno-missing-field-initializers -Wno-unused-variable+} -Werror -Wno-unused-parameter -Wno-unused-local-typedefs -Wno-unknown-pragmas")
elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
endif()
