diff --git a/crypto/md2/md2.c b/crypto/md2/md2.c
index f4d6f62..bc571d6 100644
--- a/crypto/md2/md2.c
+++ b/crypto/md2/md2.c
@@ -58,7 +58,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <openssl/md2.h>
+#include <md2/md2.h>
 
 #define BUFSIZE	1024*16
 
diff --git a/crypto/md2/md2_dgst.c b/crypto/md2/md2_dgst.c
index c57b3da..e8defcb 100644
--- a/crypto/md2/md2_dgst.c
+++ b/crypto/md2/md2_dgst.c
@@ -59,7 +59,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <openssl/md2.h>
+#include <md2/md2.h>
 #include <openssl/opensslv.h>
 #include <openssl/crypto.h>
 
diff --git a/crypto/md2/md2_one.c b/crypto/md2/md2_one.c
index f7fef5c..e5b88e4 100644
--- a/crypto/md2/md2_one.c
+++ b/crypto/md2/md2_one.c
@@ -58,7 +58,7 @@
 
 #include <stdio.h>
 #include "cryptlib.h"
-#include <openssl/md2.h>
+#include <md2/md2.h>
 
 /* This is a separate file so that #defines in cryptlib.h can
  * map my MD functions to different names */
diff --git a/crypto/md2/md2test.c b/crypto/md2/md2test.c
index db5f5bc..0baf8e7 100644
--- a/crypto/md2/md2test.c
+++ b/crypto/md2/md2test.c
@@ -70,7 +70,7 @@ int main(int argc, char *argv[])
 }
 #else
 #include <openssl/evp.h>
-#include <openssl/md2.h>
+#include <md2/md2.h>
 
 #ifdef CHARSET_EBCDIC
 #include <openssl/ebcdic.h>
diff --git a/crypto/evp/m_md2.c b/crypto/evp/m_md2.c
index 5ce849f..f42de25 100644
--- a/crypto/evp/m_md2.c
+++ b/crypto/evp/m_md2.c
@@ -64,7 +64,7 @@
 #include <openssl/evp.h>
 #include <openssl/objects.h>
 #include <openssl/x509.h>
-#include <openssl/md2.h>
+#include <md2/md2.h>
 #ifndef OPENSSL_NO_RSA
 #include <openssl/rsa.h>
 #endif
diff --git a/crypto/rand/rand_lcl.h b/crypto/rand/rand_lcl.h
index 618a8ec..ece1a89 100644
--- a/crypto/rand/rand_lcl.h
+++ b/crypto/rand/rand_lcl.h
@@ -148,7 +148,7 @@
 #define MD_Init(a)		EVP_DigestInit_ex(a,EVP_mdc2(), NULL)
 #define	MD(a,b,c)		EVP_Digest(a,b,c,NULL,EVP_mdc2(), NULL)
 #elif defined(USE_MD2_RAND)
-#include <openssl/md2.h>
+#include <md2/md2.h>
 #define MD_DIGEST_LENGTH	MD2_DIGEST_LENGTH
 #define MD_Init(a)		EVP_DigestInit_ex(a,EVP_md2(), NULL)
 #define	MD(a,b,c)		EVP_Digest(a,b,c,NULL,EVP_md2(), NULL)
diff --git a/crypto/md2/md2test.c b/crypto/md2/md2test.c
index 0baf8e7..db5f5bc 100644
--- a/crypto/md2/md2test.c
+++ b/crypto/md2/md2test.c
@@ -70,7 +70,7 @@ int main(int argc, char *argv[])
 }
 #else
 #include <openssl/evp.h>
-#include <md2/md2.h>
+#include <openssl/md2.h>
 
 #ifdef CHARSET_EBCDIC
 #include <openssl/ebcdic.h>
