From bfce4e5d6ecd445e6e000387deb9afc435778c6b Mon Sep 17 00:00:00 2001
From: Matt Caswell <matt@openssl.org>
Date: Tue, 27 May 2014 00:26:55 +0100
Subject: [PATCH] Fixed Windows compilation failure

---
 ssl/s3_pkt.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/ssl/s3_pkt.c b/ssl/s3_pkt.c
index c087fee..b516b6f 100644
--- a/ssl/s3_pkt.c
+++ b/ssl/s3_pkt.c
@@ -110,6 +110,7 @@
  */
 
 #include <stdio.h>
+#include <limits.h>
 #include <errno.h>
 #define USE_SOCKETS
 #include "ssl_locl.h"
-- 
1.7.9.5
